tests/integration_tests/performance/test_network_tcp_throughput.py

```diff
@@ -0,0 +1,396 @@
+        "tcp-ps128K-ws64K-1parallel-h2g": 7500,
+        "tcp-ps128K-ws128K-1parallel-h2g": 10500,
+        "tcp-ps128K-ws256K-1parallel-h2g": 10500,
+        "tcp-ps256K-ws10K-1parallel-h2g": 1500,
+        "tcp-ps256K-ws64K-1parallel-h2g": 7500,
+        "tcp-ps256K-ws128K-1parallel-h2g": 10500,
+        "tcp-ps256K-ws256K-1parallel-h2g": 12500,
+        "tcp-ps512K-ws10K-1parallel-h2g": 1500,
+        "tcp-ps512K-ws64K-1parallel-h2g": 7500,
+        "tcp-ps512K-ws128K-1parallel-h2g": 10500,
+        "tcp-ps512K-ws256K-1parallel-h2g": 12500,
+        "tcp-ps1024K-ws10K-1parallel-h2g": 1500,
+        "tcp-ps1024K-ws64K-1parallel-h2g": 7500,
+        "tcp-ps1024K-ws128K-1parallel-h2g": 10500,
+        "tcp-ps1024K-ws256K-1parallel-h2g": 12500,
+        "tcp-ps128K-ws10K-2parallel-bd": 2000,
+        "tcp-ps128K-ws64K-2parallel-bd": 10500,
+        "tcp-ps128K-ws128K-2parallel-bd": 13500,
+        "tcp-ps128K-ws256K-2parallel-bd": 16500,
+        "tcp-ps256K-ws10K-2parallel-bd": 2000,
+        "tcp-ps256K-ws64K-2parallel-bd": 10500,
+        "tcp-ps256K-ws128K-2parallel-bd": 13500,
+        "tcp-ps256K-ws256K-2parallel-bd": 16500,
+        "tcp-ps512K-ws10K-2parallel-bd": 2000,
+        "tcp-ps512K-ws64K-2parallel-bd": 10500,
+        "tcp-ps512K-ws128K-2parallel-bd": 13500,
+        "tcp-ps512K-ws256K-2parallel-bd": 16500,
+        "tcp-ps1024K-ws10K-2parallel-bd": 2000,
+        "tcp-ps1024K-ws64K-2parallel-bd": 10500,
+        "tcp-ps1024K-ws128K-2parallel-bd": 14500,
+        "tcp-ps1024K-ws256K-2parallel-bd": 15500,
+        "tcp-ps128K-ws10K-1parallel-bd": 1500,
+        "tcp-ps128K-ws64K-1parallel-bd": 7500,
+        "tcp-ps128K-ws128K-1parallel-bd": 10500,
+        "tcp-ps128K-ws256K-1parallel-bd": 10500,
+        "tcp-ps256K-ws10K-1parallel-bd": 1500,
+        "tcp-ps256K-ws64K-1parallel-bd": 7500,
+        "tcp-ps256K-ws128K-1parallel-bd": 10500,
+        "tcp-ps256K-ws256K-1parallel-bd": 12500,
+        "tcp-ps512K-ws10K-1parallel-bd": 1500,
+        "tcp-ps512K-ws64K-1parallel-bd": 7500,
+        "tcp-ps512K-ws128K-1parallel-bd": 10500,
+        "tcp-ps512K-ws256K-1parallel-bd": 12500,
+        "tcp-ps1024K-ws10K-1parallel-bd": 1500,
+        "tcp-ps1024K-ws64K-1parallel-bd": 7500,
+        "tcp-ps1024K-ws128K-1parallel-bd": 10500,
+        "tcp-ps1024K-ws256K-1parallel-bd": 12500,
+    },
+    "baseline_guest_cpu_utilization" : 30,
+    # Percentage
+    # "baseline_guest_cpu_utilization": {
+    #     "tcp-ps128K-ws10K-1parallel-g2h": {"target": 8, "delta": 1},
+    # }
+}
+
+IPERF3 = "iperf3"
+THROUGHPUT = "throughput"
+THROUGHPUT_TOTAL = "throughput_total"
+DURATION = "duration"
+DURATION_TOTAL = "duration_total"
+RETRANSMITS = "retransmits"
+RETRANSMITS_TOTAL = "retransmits_total"
+CPU_UTILIZATION_HOST = "cpu_utilization_host"
+CPU_UTILIZATION_HOST_TOTAL = "cpu_utilization_host_total"
+CPU_UTILIZATION_GUEST = "cpu_utilization_guest"
+CPU_UTILIZATION_GUEST_TOTAL = "cpu_utilization_guest_total"
+
+BASE_PORT = 5000
+
```

Comment:
While the tests are ok, they are missing a point - how does the emulation perform in a high throughput scenarios.
Do these tests fill up the emulation? How does the performance look when you generate a lot of contention?
I recommend to run tests with different vm sizes - 1vcpus, 2, 4, 8, 16 and also try multiple iperfs processes for each vm type 1, 2, 4, 8, 16, 128. 
Also after this PR please try with multiple rate-limiter configs, so we can know the penalty added by the rate-limiter.

Reply:
Trying with different uVMs machine configs definitely makes sense. Adding these scenarios is straight forward, starting from this test, but I suggest we keep this as it is right now, following up with uVMs of 4, 8, 16, 128 vCPUs in the future. As we have seen for what we have right now, getting baselines for these uVMs take some time. Once this PR is merged, it is just a matter of adding a new microvm artifact and getting the baselines. WDYT?

Reply:
Please track this remaining requirement